doc/src/manual/arrays.md

```diff
@@ -23,11 +23,11 @@ sharing](https://en.wikipedia.org/wiki/Evaluation_strategy#Call_by_sharing)
 while this prevents accidental modification by callees of a value in the caller,
 it makes avoiding unwanted copying of arrays difficult. By convention, a
 function name ending with a `!` indicates that it will mutate or destroy the
-value of one or more of its arguments. Callees must make explicit copies to
-ensure that they don't modify inputs that they don't intend to change. Many non-
-mutating functions are implemented by calling a function of the same name with
-an added `!` at the end on an explicit copy of the input, and returning that
-copy.
+value of one or more of its arguments (see, for example, [`sum`](@ref) and [`sum!`](@ref).
```

Comment:
I like the example pair, but I think `sum` is a bit of a strange case since it takes two differently-sized arguments and determines its operation based upon their sizes.  It'd be nice if we could use a simpler example here where the non-bang function dispatches to the bang-function of the copied arg. How about `sort` and `sort!` instead?